library/hawkular_alerts_group_trigger.py

```diff
@@ -134,6 +147,30 @@ def delete_group_trigger(self, group_id):
         except Exception as e:
             self.module.fail_json(msg="Failed to delete group trigger. Error: {error}".format(error=e))
 
+    def conditions_update_required(self, trigger_id, conditions):
+        """ Returns True if an update is required in one or more of the group
+            trigger conditions, False otherwise
+        """
+        tgs = self.client.get_trigger_conditions(trigger_id)
+
+        # compare number of conditions
+        if len(conditions) != len(tgs):
+            return True
+
+        # sort conditions lists by the condition name
+        tgs_sorted = sorted(tgs, key=lambda k: k.context['name'])
+        conditions_sorted = sorted(conditions, key=lambda k: k['name'])
+
+        for i in range(len(conditions_sorted)):
+            if conditions_sorted[i]['name'] != tgs_sorted[i].context['name']:
+                return True
+            keys_to_compare = conditions_sorted[i].keys()
+            keys_to_compare.remove('name')
+            for key in keys_to_compare:
+                if conditions_sorted[i][key] != getattr(tgs_sorted[i], key):
+                    return True
+        return False
```

Comment:
You're ignoring existing Condition's fields that weren't provided to module.
That's deliberate for fields server adds.  But are there any user-provided fields that are optional?

There is a class weak idempotency fails I see here:
where a difference between actual and desired is "small enough" that by itself it doesn't cause updating,
but IF we do update due to other reasons, this difference will be blown away (because all conditions are re-set from scratch, right?).
Another example is non-'name' context fields — it's good we allow them but we will destroy them on update.
(Another is allowing omitted conditions (L274) which doesn't immediately delete the conditions, but would not re-create them.)

These scenarios themselves are not deal-breakers, and maybe defensible as features.  I just worry the cumulative idempotency behavior becomes too complex to understand or document...